opm/simulators/utils/ParallelNLDDPartitioningZoltan.cpp

```diff
@@ -0,0 +1,607 @@
+    ///   \c VertexGraph instance.
+    ///
+    /// \param[in] numElmsPerGid Number of ZOLTAN_ID_TYPE elements needed to
+    ///   describe a single globally unique object/vertex ID.  Must be one
+    ///   (1) in this implementation.
+    ///
+    /// \param[in] numElmsPerLid Number of ZOLTAN_ID_TYPE elements needed to
+    ///   describe a single local object/vertex ID.  Must be one (1) in this
+    ///   implementation.
+    ///
+    /// \param[in] numCells Number of cells/vertices/objects for which to
+    ///   identify the neighbouring cells/vertices/objects.
+    ///
+    /// \param[in] localIds Local object/vertex IDs.  Size equal to \code
+    ///   numElmsPerLid * numVertices(graphPtr) \endcode.
+    ///
+    /// \param[in,out] neighbourGid Globally unique object ID of each
+    ///   neighbour cell/vertex/object.  Populated by this function.
+    ///   Allocated by Zoltan.
+    ///
+    /// \param[in,out] neighbourProc Owner of each neighbouring
+    ///   cell/vertex/object.  Populated by this function.  Allocated by
+    ///   Zoltan.
+    ///
+    /// \param[out] ierr Error code for Zoltan consumption.  Single \c int.
+    void edgeList(void*            graphPtr,
+                  const int        numElmsPerGid,
+                  const int        numElmsPerLid,
+                  const int        numCells,
+                  ZOLTAN_ID_PTR /* globalIds */,
+                  ZOLTAN_ID_PTR    localIds,
+                  int*          /* numEdges */,
+                  ZOLTAN_ID_PTR    neighbourGid,
+                  int*             neighbourProc,
+                  int           /* weightDim */,
+                  float*        /* edgeWeights */,
+                  int*             ierr)
+    {
+        const auto* graph = static_cast<VertexGraph*>(graphPtr);
+
+        if ((numElmsPerGid != numElmsPerLid) ||
+            (numElmsPerLid != 1) ||
+            (numCells != graph->numVertices()))
+        {
+            *ierr = ZOLTAN_FATAL;
+        }
+
+        if (graph->numVertices() == 0) {
+            *ierr = ZOLTAN_OK;
+            return;
+        }
+
+        const auto& ia = graph->startPointers();
+        const auto& ja = graph->columnIndices();
+
+        for (auto cell = 0*numCells, ix = 0; cell < numCells; ++cell) {
+            const auto localCell = localIds[cell];
+            for (auto neighIx = ia[localCell], end = ia[localCell + 1];
+                 neighIx != end; ++neighIx, ++ix)
+            {
+                neighbourGid [ix] = static_cast<ZOLTAN_ID_TYPE>(graph->globalId(ja[neighIx]));
+                neighbourProc[ix] = graph->myRank();
+            }
+        }
+    }
+} // extern "C"
+
+    /// Partition VertexGraph objects using Zoltan graph partitioning software.
+    class Partitioner
+    {
+    public:
+        /// Constructor.
+        ///
+        /// \param[in] comm MPI communicator.
+        ///
+        /// \param[in] params Control parameters for Zoltan graph partiitioning procedure.
```

Comment:
partitioning

Reply:
Thanks for noticing. I've pushed an update to fix that.